mreg/api/v1/tests/test_filtering.py

```diff
@@ -0,0 +1,109 @@
+from typing import List
+from itertools import chain
+
+from unittest_parametrize import ParametrizedTestCase, param, parametrize
+
+from mreg.models.host import Host, Ipaddress
+from mreg.models.resource_records import Cname
+
+from .tests import MregAPITestCase
+
+def create_hosts(name: str, count: int) -> List[Host]:
+    """Create hosts."""
+
+    hosts: List[Host] = []
+    for i in range(count):
+        hosts.append(Host.objects.create(
+            name=f"{name}{i}.example.com".replace("_", ""),
+            contact="admin@example.com",
+            ttl=3600,
+            comment="Test host",
+        ))
+
+    return hosts
+
+def create_cnames(hosts: List[Host]) -> List[Cname]:
+    """Create cnames."""
+
+    cnames: List[Cname] = []
+    for host in hosts:
+        cnames.append(Cname.objects.create(
+            host=host,
+            name=f"cname.{host.name}",
+            ttl=3600,
+        ))
+
+    return cnames
+
+def create_ipaddresses(hosts: List[Host]) -> List[Ipaddress]:
+    """Create ipaddresses."""
+
+    ipaddresses: List[Ipaddress] = []
+    for i, host in enumerate(hosts):
+        ipaddresses.append(Ipaddress.objects.create(
+            host=host,
+            ipaddress=f"10.0.0.{i}",
+        ))
+
+    return ipaddresses
+
+class FilterTestCase(ParametrizedTestCase, MregAPITestCase):
+    """Test filtering."""
+
+    # endpoint, query_key, target, expected_hits
+    #
+    # NOTE: The generated hostnames are UNIQUE across every test case!
+    # The format is: f"{endpoint}{query_key}{i}.example.com".replace("_", "")
+    # where i is the index of the hostname (and we make three for each test).
+    @parametrize(
```

Comment:
Should there be a test for `atoms__name__exact` here, or is that unnecessary?

Reply:
There should be a set of tests for policy. I'll add that.

Reply:
Fixed in https://github.com/unioslo/mreg/pull/547/commits/99fee8a2d5a58931d2cbf7bc4bf79b36dcd36b98.